code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -749,3 +749,83 @@
 			D.set_color()
 			D.set_up(my_target, rand(6,8), 1, 50)
 	return
+
+/mob/living/carbon/human/proc/thunder()
+	set category = "Abilities"
+	set name = "Thunderbolt"
+	set desc = "Release your inner electricity, creating a powerful discharge of lightning."
+
+	if(last_special > world.time)
+		return
+
+	if(stat || paralysis || stunned || weakened || lying)
+		to_chat(src,"<span class='warning'>You cannot do that in your current state!</span>")
+		return
+
+	visible_message("<span class='danger'>\The [src] crackles with energy!</span>")
+
+	playsound(src, 'sound/magic/LightningShock.ogg', 75, 1)
+
+	tesla_zap(src, 7, 1500)
+
+	last_special = world.time + 50
+
+/mob/living/carbon/human/proc/consume_material()
+	set category = "Abilities"
+	set name = "Incorporate Matter"
+	set desc = "Repair your damage body by using the same materials you were made from."
+
+	if(last_special > world.time)
+		return
+
+	if(stat || paralysis || stunned || weakened || lying)
+		to_chat(src,"<span class='warning'>You cannot do that in your current state!</span>")
+		return
+
+	var/obj/item/I = src.get_active_hand()
+
+	if(I)
+		if(istype(I, /obj/item/stack/material))
+			var/obj/item/stack/material/O = I
+			if(O.material.golem == src.species.name)
```

Comment:
yee this is about to end horribly. Isn't it better to make `material/var/golem` contain a species _type_? This is provided that all golem species are unique type paths. But, if they are, then ye make the var hold a species datum type and run `if (istype(src.species, O.material.golem))`